vm/devices/vmbus/vmbus_server/src/lib.rs

```diff
@@ -231,6 +231,11 @@ impl EventPort for ChannelEvent {
 pub struct SavedState {
     #[mesh(1)]
     server: channels::SavedState,
+    // Indicates if the lost synic bug is fixed or not. By default it's false.
+    // During the restore process, we check if the field is not true then
+    // unstick_channels() function will be called to mitigate the issue.
+    #[mesh(2)]
+    lost_synic_bug_fixed: bool,
```

Comment:
Can we use a version number here instead so we can increment it whenever we need to detect behavior changes like this, instead of adding additional bools. So '0' would mean before the version field existed, this could be '1' or '2' now depending on preference, and eventually for the next compat issue it would increment again.

Reply:
I agree with this. @jstarks, any objections?

Reply:
Version numbers imply a linear progression of fixes/changes, which is not always the case due to backports. E.g, if we get to version 5 with various fixes and then want to backport just the latest fix to an older release, do we need to backport all the fixes?

Reply:
A bitfield, then? Or does that not offer enough advantages over booleans (I don't know how protobuf packs the values).

Reply:
It's not clear that there's much benefit to that at the protobuf level. Each present bool will take up 2 bytes in the protobuf encoding. That doesn't seem like a problem.

Reply:
Okay, I'm fine with leaving it as is, then.